### PR TITLE
Update to latest version of our otel library fork.

### PR DESCRIPTION
**Pull request by umanwizard**
This entails a refactor of how we get kernel frames -- these are now just communicated via FrameMetadata, the same was as normal frames.

### CODE REVIEW

File: reporter/arrow.go [hunk 1]
```diff
@@ -36,6 +36,14 @@ func (b *BinaryDictionaryRunEndBuilder) NewArray() arrow.Array {
...
 }
 
+func (b *BinaryDictionaryRunEndBuilder) Value(n int) []byte {
+	if b.idx.Len() > 0 &&
+		!b.idx.IsNull(b.idx.Len()-1) {
+		return b.bd.Value(int(b.idx.Value(b.idx.Len() - 1)))
+	}
+	return nil
+}
+
```
Comment (umanwizard) [inline reporter/arrow.go:46]:
```suggestion
```

File: go.mod [hunk 2]
```diff
@@ -149,4 +149,4 @@ require (
 	sigs.k8s.io/yaml v1.3.0 // indirect
 )
 
-replace github.com/open-telemetry/opentelemetry-ebpf-profiler => github.com/parca-dev/opentelemetry-ebpf-profiler v0.0.0-20241009170523-4d0512c029e0
+replace go.opentelemetry.io/ebpf-profiler => ../opentelemetry-ebpf-profiler
```
Comment (gnurizen) [inline go.mod]: Scapel left in patient?

File: reporter/parca_reporter.go [hunk 3]
```diff
@@ -240,87 +235,92 @@ func (r *ParcaReporter) ReportFramesForTrace(_ *libpf.Trace) {}
...
+	r.executables.Add(args.FileID, metadata.ExecInfo{
+		FileName: args.FileName,
+		BuildID:  args.GnuBuildID,
 		Compiler: ainur.Compiler(ef),
 		Static:   ainur.Static(ef),
 		Stripped: ainur.Stripped(ef),
 	})
 }
 
+// FrameKnown is a no-op for ParcaReporter.
```
Comment (gnurizen) [inline reporter/parca_reporter.go:279]: stale comment

File: reporter/parca_reporter.go [hunk 4]
```diff
@@ -656,6 +650,9 @@ func (r *ParcaReporter) reportDataToBackend(ctx context.Context, buf *bytes.Buff
 	}
 
 	rec, err = r.buildStacktraceRecord(ctx, stacktraceIDs)
+	// i := rec.NumRows()
+	// fmt.Printf("stacktrace record has %d rows. Col3: %s\n", i, rec.Column(3).String())
+
```
Comment (gnurizen) [inline reporter/parca_reporter.go:655]: Scapel left in patient

**Review by gnurizen — APPROVED** [review]

**Review by gnurizen — APPROVED** [review]

**Comment by gnurizen** [comment]
Looks good, just some nits to clean up

File: reporter/parca_reporter.go (hunk 4 above)
Comment (umanwizard) [reply reporter/parca_reporter.go:655]: 🤦‍♂️

**Pull request merged**